cloud_governance/cloud_resource_orchestration/aws/ec2/cost_over_usage.py

```diff
@@ -0,0 +1,266 @@
+        :return:
+        """
+        over_usage_users = []
+        current_month_users = self.get_monthly_user_es_cost_data()
+        for user in current_month_users:
+            user_name = user.get('User')
+            cost = round(user.get('Cost'), self.DEFAULT_ROUND_DIGITS)
+            if cost >= (self.__over_usage_amount - self.__over_usage_threshold):
+                query = {  # check user opened the ticket in elastic_search
+                    "query": {
+                        "bool": {
+                            "must": {
+                                "term": {"user": user_name}
+                            },
+                            "filter": {
+                                "range": {
+                                    "timestamp": {
+                                        "format": "yyyy-MM-dd",
+                                        "lte": str(self.current_end_date.date()),
+                                        "gte": str(self.current_start_date.date()),
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                response = self.es_operations.fetch_data_by_es_query(es_index=self.es_index_cro, query=query)
+                if not response:
+                    over_usage_users.append(user)
+                for cro_data in response:
+                    project = cro_data.get('_source').get('project')
+                    ticket_id_state = cro_data.get('_source').get('ticket_id_state')
+                    if 'close' in ticket_id_state.lower():
+                        ticket_closed_time = datetime.strptime(cro_data.get('_source').get('timestamp'), "%Y-%m-%dT%H:%M:%S.%f")
+                        month_start_date = ticket_closed_time if self.current_start_date < ticket_closed_time else self.current_start_date
+                        user_response = self.get_monthly_user_es_cost_data(start_date=month_start_date, extra_matches={'Tags': {'Key': 'User', 'Values': [user_name]}}, extra_operation='And', tag_name='Project')
+                        if user_response:
+                            if project in [res.get('Project') for res in user_response]:
+                                user_cost = round(user_response[0].get('Cost'), self.DEFAULT_ROUND_DIGITS)
+                                user['Cost'] = user_cost
+                                user['Project'] = project
+                                if user_cost >= self.__over_usage_amount:
+                                    over_usage_users.append(user)
+                    elif 'progress' in ticket_id_state.lower():
+                        opened_ticket_cost = cro_data.get('_source').get('estimated_cost')
+                        if cost - opened_ticket_cost > self.__over_usage_amount:
+                            over_usage_users.append(user)
```

Comment:
@ebattat  Here, I added the condition.

Reply:
Need to add method in_progress_user_cost() and then 
user_cost - opened_ticket_cost > self.__over_usage_amount:

Reply:
If cost is user cost, pls change it to user_cost